src/linked_list.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+"""This file implements a basic linked list data structure."""
+from __future__ import unicode_literals
+
+
+class Node(object):
+    """This builds our Node class."""
+
+    def __init__(self, data):
+        """Initial setup of Node objects."""
+        self.data = data
+        self.next_node = None
+
+
+class LinkedList(object):
+    """This builds LinkedList class."""
+
+    def __init__(self, data=None):
+        """Initial setup of LinkedList class."""
+        self.head = None
+        try:
```

Comment:
This `try...except` block deals entirely with the possibility of having a value passed for `data` which is not iterable, right?  If so, then it belongs *inside* the if block on line 22, not outside.